core/src/main/scala/kafka/server/DynamicConfig.scala

```diff
@@ -36,6 +36,7 @@ object DynamicConfig {
     //Properties
     val LeaderReplicationThrottledRateProp = "leader.replication.throttled.rate"
     val FollowerReplicationThrottledRateProp = "follower.replication.throttled.rate"
+    val IntraBrokerReplicationThrottledRateProp = "intra.broker.replication.throttled.rate"
```

Comment:
Is `log.cleaner.io.max.bytes.per.second` the config for throttling log cleaning? I am not sure it is easy to be confused with `intra.broker.replication.throttled.rate`. The former uses `log.cleaner` in the name thus we know it is for log cleaner. The latter uses `replication` in the name and therefore we know it is for replication within the broker. Did I miss something here?

Reply:
Seems like we made a bit of a mistake with the existing throttling configs in that they don't specify the unit. The log cleaner setting gets that right. Also, it seems like `replication` should be reserved for data transfer between two replicas. Are there two replicas here or is this just a case of copying data from one log dir to another?

Reply:
I was thinking that longer term, it may be better to have a single quota that covers all I/Os within a broker (including log cleaner, moving data across disks, etc). In that case, a config name with "intra.broker" may be more appropriate. However, if we use that name for moving data across disks, it may make future naming a bit harder.

Reply:
@ijuma It is two replicas here instead of simply copying data from one log dir to another because we are maintaining states such as epoch and high watermark in the future replica.

@junrao I think it is a good idea to have a single config to throttle all I/Os within a broker. Given that this probably needs to deprecate the existing `log.cleaner.io.max.bytes.per.second` config, maybe we should do it in a separate KIP and find a good name for it?

Regarding this patch, I think `intra.broker.replication.throttled.rate` is OK because this name says "replication" which means fetching data from one replica to another. I am also good with `across.dirs.replication.throttled.rate` (which is more consistent with `leader.replication.throttled.rate`). Do you prefer me to change the name to `across.dirs.replication.throttled.rate`?

Reply:
How about log.dirs.balancing.io.max.bytes.per.second ?

Reply:
Sure. I have updated the patch to rename this config and related variables as suggested.